src/posts/posts.repository.ts

```diff
@@ -50,8 +57,9 @@ export class PostsRepository extends Repository<PostSocial> {
   async updatePost(
     id: string,
     updatePostDto: ActionPostDto,
+    user: User,
   ): Promise<PostSocial> {
-    const post = await this.getPostById(id);
+    const post = await this.getPostById(id, user);
```

Comment:
Em nên tận dụng các relationship đã khai báo để gọi posts của user thông qua user nhé
Sẽ gọi kiểu user.posts để lấy được các posts của user thay vì query post truyền vào 2 tham số như này
Nếu lấy cụ thể post ID của user thì sẽ mới cần query thêm ID như là user.posts.findOne(id: ID).

Reply:
Dạ e hiểu ý của a nhưng e k biết triển khai ở code ra sao ạ. A hướng dẫn giúp e với ạ.

Reply:
Phần này em triển khai như hiện tại hỉ, 
trường hợp cần dùng đến relation thì cần thông qua user repository của user, ở đây cũng chưa có cần thiết

em tham khảo thêm: https://github.com/typeorm/typeorm/blob/master/docs/many-to-one-one-to-many-relations.md

Reply:
Em cứ chạy bình thường để hoàn thiện các task nha.
Sau này a sẽ support refactor lại để m cũng có thể học thêm được về cái đó